vllm/config.py

```diff
@@ -2886,9 +2886,11 @@ def get_served_model_name(model: str,
 class DecodingConfig:
     """Dataclass which contains the decoding strategy of the engine"""
 
+    DEFAULT_BACKEND: str = "auto" if envs.VLLM_USE_V1 else "xgrammar"
+
     # Which guided decoding algo to use.
     # 'outlines' / 'lm-format-enforcer' / 'xgrammar'
-    guided_decoding_backend: str = 'xgrammar'
+    guided_decoding_backend: str = DEFAULT_BACKEND
```

Comment:
Oh and instead of the comment you could type hint it with `Literal["auto", "outlines", "lm-format-enforcer", "xgrammar"]`. It makes the line longer, but it's very IDE friendly.

Reply:
It's tricky because the valid set of values is different between V0 and V1.